src/main/resources/common-services/CDAP/5.0.0/package/scripts/master.py

```diff
@@ -57,6 +60,62 @@ def start(self, env, upgrade_type=None):
             not_if=no_op_test
         )
 
+    def killOlderProcesses(self):
+        # execute ls /cdap/election/master.services and check if emty then only execute the following steps else exit
+
+        import params
+        Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " ls /election/master.services > /tmp/election_master_services", user='zookeeper')
```

Comment:
Hardcoding Zookeeper path '/usr/hdp/3.1.0.0-78/zookeeper/' should be avoided. Instead, use '/usr/hdp/current/' to make it compatible with multiple HDP versions.